### DISCUSSION TITLE
How exactly does groups work? How do I keep thing private?

### DISCUSSION THREAD

**Discussion by RefineryX** [mealie-recipes/mealie#2401] [Q&A]
I am struggling with the concept of Groups. 

On the [homepage](https://nightly.mealie.io), it describes it as a way to "Sort users into groups to share recipes and mealplans while keeping your cooking club separate." How do I keep my cooking separate? Everytime I add a recipe or cookbook, all users of the group can see it. Likewise, if a user adds something it automatically gets added to mine.

**User Case:** I would like users to keep their recipes private but also a way to contribute and view recipes in a 'shared' space.

**Comment by hay-kot** [comment]
> "Sort users into groups to share recipes and mealplans while keeping your cooking club separate."

I'm not really sure why it says that. I don't remember writing it :shrug:

> I would like users to keep their recipes private but also a way to contribute and view recipes in a 'shared' space.

Is not yet implemented. You can share individual recipes, but sharing groups isn't yet supported. See https://github.com/hay-kot/mealie/issues/1326 for more info

**Reply by RefineryX** [reply]
Thanks. After review of #1326, I think that is what I am looking for.

I am still somewhat unclear on the concept of users and groups. Would it not make it easier if groups were only used for recipe sharing? Meaning that:

**User accounts are private by default**
1. Users are created without requiring them to be part of a "group"
2. User recipes not shared outside their account
3. Any recipes added are owned by the user account

**Sharing Abilities**
When the user would like to share certain recipes:

4. They can create a shared space / section such as a "cookbook" or a "group" 
5. Only recipes they add to the shared space are shared
6. All their other recipes, settings and cookbooks etc remain private for the user


**Example** 
Husband and wife have two accounts. They are able to both add their recipes within their accounts which are kept private. They also created a **shared space** for recipes they would like to try / cook together.

**Reply by hay-kot** [reply]
@RefineryX 

The sharing model is difficult because there's different parts a the app that would need varying levels of sharing, Recipes, Meal Plans, Cookbooks, Shopping Lists, etc...

I don't think we have the best model, but it would be a pretty big lift to migrate to a new one, and I don't think anyone has the appetite to take on the technical load right now. 

This is the most recent ideas I've had around this, it hasn't been implemented though.

https://github.com/hay-kot/mealie/discussions/2134
